source/extensions/filters/http/ext_proc/ext_proc.cc

```diff
@@ -512,6 +541,16 @@ void Filter::onReceiveMessage(std::unique_ptr<ProcessingResponse>&& r) {
     closeStream();
     clearAsyncState();
     processing_complete_ = true;
+  } else {
+    // Any other error results in an immediate response with an error message.
+    ENVOY_LOG(debug, "Sending immediate response: {}", processing_status.message());
+    processing_complete_ = true;
+    closeStream();
+    cleanUpTimers();
+    ImmediateResponse invalid_mutation_response;
+    invalid_mutation_response.mutable_status()->set_code(StatusCode::InternalServerError);
+    invalid_mutation_response.set_details(std::string(processing_status.message()));
+    sendImmediateResponse(invalid_mutation_response);
```

Comment:
Should we be incrementing a stat for this pathway? Or is that already covered?

Reply:
Yes, I think we should increment the "stream_msgs_received" counter here, so I fixed that. Thanks!